### PR TITLE
Add IOU badge to report row and header

### PR DESCRIPTION
**Pull request by marcaaron**
### Details
Adds IOU badge.

**Notes:**
- We still aren't yet handling the Pusher event so once the IOU is paid things do not update in realtime. This can probably be fixed in a later PR.
- The report header text wraps weirdly now that there is an extra element. I forced it to use ellipsis instead since I think it looks better.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/158424

### Tests
1. Create a new IOU transaction via
```
CreateIOUTransaction
authToken: <auth token>
debtorEmail: <email of user you want to request payment from>
currency: USD
comment: 'test'
amount: 299
```
2. Open E.cash as the user (same authToken that you just used)
3. Verify that there is now a badge to in the report row and header
4. Run the next command to mark it as paid
```
PayIOU
authToken: <auth token of the payee> // must sign in as the "debtor"
reportID: <IOU Report ID> 
paymentMethodType: 'Venmo'
```
5. Refresh the page
6. Verify that the badges are no longer displayed

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
Screen Shot 2021-03-26 at 7 57 47 AM

#### Mobile Web
Screen Shot 2021-03-26 at 8 09 37 AM
Screen Shot 2021-03-26 at 8 09 17 AM

#### Desktop
Screen Shot 2021-03-26 at 8 11 58 AM

#### iOS
Screen Shot 2021-03-26 at 8 01 16 AM
Screen Shot 2021-03-26 at 8 01 06 AM

#### Android
Screen Shot 2021-03-26 at 8 07 59 AM
Screen Shot 2021-03-26 at 8 07 50 AM

### CODE REVIEW

File: src/styles/styles.js [hunk 1]
```diff
@@ -228,23 +228,18 @@ const styles = {
     pill: {
```
Comment (shawnborton) [inline src/styles/styles.js:228]: I think it might be worth renaming this to badge

File: src/styles/styles.js [hunk 2]
```diff
@@ -228,23 +228,18 @@ const styles = {
...
         borderRadius: 14,
         backgroundColor: themeColors.pillBG,
```
Comment (shawnborton) [inline src/styles/styles.js:230]: I think we'll also have a gray badge that will be the default case, so we might want to think about how to manage the styles here. Maybe we have `badge` as well as `badge.success` or something? Similar to how we handle default buttons versus success buttons? 

Or the default badge is green, and then we can also have `badge.neutral` or something?

File: src/components/IOUBadge.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+const IOUBadge = props => (
+    <TouchableOpacity
+        style={[styles.pill, styles.ml2]}
+    >
+        <Text
+            style={styles.pillText}
+            numberOfLines={1}
+        >
+            {`$${Num.number_format(props.iouReport.total / 100, 2)}`}
```
Comment (Julesssss) [inline src/components/IOUBadge.js:31]: NAB, but this looks like something we'll be reusing.

Comment (Julesssss) [reply src/components/IOUBadge.js:31]: Specifically, the total > dollar format

File: src/styles/themes/default.js [hunk 4]
```diff
@@ -29,7 +29,7 @@ export default {
     sidebarButtonBG: 'rgba(198, 201, 202, 0.25)',
     modalBackdrop: colors.gray3,
     modalBackground: colors.gray2,
-    pillBG: colors.gray2,
+    pillBG: colors.green,
```
Comment (Julesssss) [inline src/styles/themes/default.js:32]: This too -- if we end up changing the style name

File: src/styles/styles.js [hunk 5]
```diff
@@ -228,23 +228,18 @@ const styles = {
...
-        height: variables.componentSizeSmall,
+        height: 20,
         flexDirection: 'row',
-        paddingTop: 6,
-        paddingBottom: 6,
         paddingLeft: 7,
         paddingRight: 7,
```
Comment (Julesssss) [inline src/styles/styles.js:234]: A couple of minor changes to make [according to the Figma design](https://www.figma.com/file/hXumnAqZUEULJcP8SFnvYT/NL5%3A-Request-Money-%26-Split-Bills?node-id=128%3A0):
- BadgeHeight should be 26
- HorizontalPadding should be 8

File: src/components/IOUBadge.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import PropTypes from 'prop-types';
+import {Text, TouchableOpacity} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import Num from 'expensify-common/lib/Num';
+import ONYXKEYS from '../ONYXKEYS';
+import styles from '../styles/styles';
+
+const propTypes = {
+    // IOU Report data object
+    iouReport: {
```
Comment (Julesssss) [inline src/components/IOUBadge.js:11]: I think this is missing `PropType shape`

Comment (Julesssss) [reply src/components/IOUBadge.js:11]: `iouReport: PropTypes.shape({...})`

File: src/components/IOUBadge.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+const defaultProps = {
+    iouReport: {
+        total: 0,
+    },
+};
+
+const IOUBadge = props => (
+    <TouchableOpacity
+        style={[styles.pill, styles.ml2]}
+    >
```
Comment (Julesssss) [inline src/components/IOUBadge.js:26]: As we've descoped the tap to scroll feature for now I think the TouchableOpacity should also be removed.

File: src/components/IOUBadge.js (hunk 6 above)
Comment (marcaaron) [reply src/components/IOUBadge.js:11]: Ah yep I rushed this one 😅

**Review by Julesssss — DISMISSED** [review]
Looks good, just the conflict to resolve

**Review by Julesssss — DISMISSED** [review]

**Comment by marcaaron** [comment]
Updated screens lmk if this is feeling right @shawnborton !

**Comment by shawnborton** [comment]
Hmm it doesn't feel quite right anymore, but previously it was looking right. Ideally the pill ends up being only 20px tall, the same exact height as the pin icon next to it.

**Comment by Julesssss** [comment]
> Hmm it doesn't feel quite right anymore, but previously it was looking right. Ideally the pill ends up being only 20px tall, the same exact height as the pin icon next to it.

@shawnborton whoops, that's my fault. I [mentioned here](https://github.com/Expensify/Expensify.cash/pull/2095#discussion_r602353326) that the size didn't match the Figma design. Was there a newer design somewhere?

**Comment by shawnborton** [comment]
Ah yes, no worries there. I think there have been a couple of different sizes floating around but most recently we landed on having the row/header badges be the same height as the pin icon.

**Comment by marcaaron** [comment]
No worries. Here's the updated screenshot for web (not gonna re-do them for all platforms since the change is pretty small)

Screen Shot 2021-03-29 at 6 53 56 AM

**Review by Julesssss — APPROVED** [review]

**Review by shawnborton — APPROVED** [review]

**Comment by shawnborton** [comment]
Nice, thanks for the update. Going to merge since Jules approved.

**Comment by Luke9389** [comment]
Hey @marcaaron, I just resolved a merge conflict that involves this PR, and want to be sure that I didn't break the functionality here. When I test this locally, should I be creating this job via /scripts/bwm.sh? If so should I just pull the `auth token` from the `session` key in local storage?
```
CreateIOUTransaction
authToken: <auth token>
debtorEmail: <email of user you want to request payment from>
currency: USD
comment: 'test'
amount: 299
```

**Comment by marcaaron** [comment]
That is an Auth command not a bedrock job.

**Comment by Luke9389** [comment]
OK cool! Thanks for steering me in the right direction.

**Pull request merged**